### ISSUE TITLE
Error: can't use Float in unions yet, use a more specific type

### ISSUE THREAD

**Issue by robacarp** [jgaskins/redis#38]
I'm running the 0.6.1 release tag under crystal 1.6.1. 

This code will produce the compiler error:

```crystal
redis.zadd "key", Time.utc.to_unix_ms, "value"
```

Perhaps because I'm [sending an Int](https://crystal-lang.org/api/1.6.2/Time.html#to_unix_ms%3AInt64-instance-method) instead of the correct type (String) the compiler is giving up? In my case I was able to add a call to #to_s and move forward, but the error seems to point to a problem in the method definition.

```
> crystal -v
Crystal 1.6.2 [879691b2e] (2022-11-03)

LLVM: 10.0.0
Default target: aarch64-apple-darwin

> shards build
Dependencies are satisfied
Building: demo
Error target demo failed to compile:
Showing last frame. Use --error-trace for full trace.

In lib/redis/src/commands/sorted_set.cr:42:43

 42 | def zadd(key : String, score : String | Float, value : String)
                                              ^----
Error: can't use Float in unions yet, use a more specific type
```

**Comment by jgaskins** [comment]
Hmm, that error message is misleading, and it changed in Crystal 1.6.0. Given this code:

```crystal
def foo(value : String | Float)
  value
end

pp foo(123)
```

When you execute it in Crystal 1.5:

```
➜  Code docker run --rm -v ~/Code:/code crystallang/crystal:1.5.1 crystal /code/foo.cr
Showing last frame. Use --error-trace for full trace.

In code/foo.cr:5:4

 5 | pp foo(123)
        ^--
Error: no overload matches 'foo' with type Int32

Overloads are:
 - foo(value : String | Float)
```

That tells you exactly what you need to know. But in Crystal 1.6:

```
➜  Code docker run --rm -v ~/Code:/code crystallang/crystal:1.6.0 crystal /code/foo.cr
Showing last frame. Use --error-trace for full trace.

In code/foo.cr:1:26

 1 | def foo(value : String | Float)
                              ^----
Error: can't use Float in unions yet, use a more specific type
```

Also, I think I wrote that method signature assuming that auto-promotion of ints to floats would work (I don't use sorted sets very rarely in Redis), but I suppose it didn't happen here because it was a union type. I can add `Int` to that signature because it *should* work (the Redis server will handle an int), but this seems like a regression in Crystal itself.

**Comment by jgaskins** [comment]
Fixed in 555eff8
